R/module_teal_with_splash.R

```diff
@@ -1,23 +1,56 @@
 # This file adds a splash screen for delayed data loading on top of teal
 
-#' UI to show a splash screen in the beginning, then delegate to [srv_teal()]
+#' `teal` module integration with splash screen
 #'
 #' @description `r lifecycle::badge("stable")`
 #'
-#' The splash screen could be used to query for a password to fetch the data.
-#' [init()] is a very thin wrapper around this module useful for end-users which
+#' Module offers the dynamic display that switches between a custom splash screen
+#' for delayed data loading and a default splash screen.This helps ensure smooth
+#' user interaction, particularly during the initial heavy data processing or
+#' when querying for a password to fetch data.
+#'
+#' @details
+#' [`init()`] is a wrapper around this module useful for end-users which
 #' assumes that it is a top-level module and cannot be embedded.
 #' This function instead adheres to the `shiny` module conventions.
```

Comment:
```suggestion
#' Add splash screen to `teal` application.
#'
#' @description `r lifecycle::badge("stable")`
#'
#' Displays custom splash screen during initial delayed data loading.
#'
#' @details
#' This module pauses app initialization pending delayed data loading.
#' This is necessary because the filter panel and modules depend on the data to initialize.
#'
#' `teal_with_splash` follows the `shiny` module convention.
#' [`init()`] is a wrapper around this that assumes that `teal` it is
#' the top-level module and cannot be embedded.
#'
#' Note it is no longer recommended to embed `teal` in `shiny` apps as a module
#' but rather use `init` to create standalone application.
```